### ISSUE TITLE
Inconsistent behaviour in fields between editors

### ISSUE THREAD

**Issue by beansmyname** [Cheeseness/icicle#3]
Clicking away from selection fields in the Scene Editor doesn't deselect that field. In the Game Editor, the text field will be deselected.

Examples:
Clicking the Game Name field in the Game Editor selects the field and makes it available for entry. Clicking anywhere outside that field, deselects and releases the cursor for selection.
Clicking the Name field in the Scene Editor locks the cursor to that field until ESC or ENTER is pressed.
Same goes for color selection in those two editors.

Using the Linux version.

**Comment by Cheeseness** [comment]
Cheers.

This is the result of me being undecided about whether to allow users to click off to remove focus from editable GUI widgets, and then being sloppy about which implementation I ended up sticking with (and the cause of that sloppiness being that the mouse input handling code is all a crazy mess of temp stuff that I'll be reworking after the event once I've got a clear picture of where I need to go next).

The key question I want to ask people before addressing this is should clicking off a field apply any edited values, or cancel them?
